tests/unit/httpd_test.cc

```diff
@@ -1997,3 +1997,64 @@ SEASTAR_THREAD_TEST_CASE(test_http_with_broken_wire) {
 
     c.close().get();
 }
+
+SEASTAR_TEST_CASE(test_client_close_connection) {
+    return async([] {
+        condition_variable cv;
+        loopback_connection_factory lcf(1);
+        auto cln = http::experimental::client(std::make_unique<loopback_http_factory>(lcf), 1, http::experimental::client::retry_requests::yes);
+
+        auto make_test_request = [&cv, &cln]() {
+            auto req = http::request::make("GET", "test", "/test");
+            cln.make_request(
+                   std::move(req),
+                   [&cv](const http::reply& resp, input_stream<char>&& in) {
+                       return seastar::async([&cv, in = std::move(in)]() mutable {
+                           auto buff = in.read().get();
+                           cv.signal();
+                       });
+                   },
+                   http::reply::status_type::ok)
+                .get();
+        };
+
+        size_t response_size = 0;
+        size_t throw_count = 0;
+        size_t nothrow_count = 0;
+        auto make_response = [&cv, &response_size, &throw_count, &nothrow_count](accept_result ar) {
+            return async([&cv, response_size, &throw_count, &nothrow_count, sk = std::move(ar.connection)]() mutable {
+                input_stream<char> in = sk.input();
+                read_simple_http_request(in);
+                output_stream<char> out = sk.output();
+                sstring r200(format("HTTP/1.1 200 OK\r\nHost: localhost\r\nContent-Length: {}\r\n\r\n", response_size));
+                out.write(r200).get();
+                out.flush().get();
+                out.write(sstring(response_size / 2, 'a')).get();
+                out.flush().get();
+                cv.wait().get();
+                if (response_size / 2 > 128 * 1024) {
+                    ++throw_count;
+                    BOOST_REQUIRE_EXCEPTION(
```

Comment:
Why will it throw?

Reply:
coz the socket is already closed when we decide on the client side not to drain the socket and just close the connection